### PR TITLE
fix(gw):  IPIP-402 CARs return useful blocks on not found errors

### PR DESCRIPTION
**Pull request by aschmahmann**
Fix extracted from #369 that could be suitable for a patch release.

### CODE REVIEW

File: gateway/gateway_test.go [hunk 1]
```diff
@@ -763,7 +763,7 @@ func TestErrorBubblingFromBackend(t *testing.T) {
 		})
 	}
 
-	testError("404 Not Found from IPLD", &ipld.ErrNotFound{}, http.StatusNotFound)
+	testError("500 Not Found from IPLD", &ipld.ErrNotFound{}, http.StatusInternalServerError)
```
Comment (aschmahmann) [inline gateway/gateway_test.go:766]: I think this was a mistake. `ipld.ErrNotFound` is an "I can't find this block"  kind of error which IIUC is supposed to be 5xx as opposed to "this path doesn't exist/IPLD errors" which are 4xx.

I think this error could also reasonably be a `502`

File: .github/workflows/gateway-conformance.yml [hunk 2]
```diff
@@ -40,7 +40,7 @@ jobs:
 
       # 4. Run the gateway-conformance tests
       - name: Run gateway-conformance tests
-        uses: ipfs/gateway-conformance/.github/actions/test@v0.3
+        uses: ipfs/gateway-conformance/.github/actions/test@261cc49391dcd2bddd7704ad28b236dc4a63c424
```
Comment (lidel) [inline .github/workflows/gateway-conformance.yml]: - [x] Change to `@v0.3` (or later)  before merge

**Review by lidel — APPROVED** [review]
I've tested this in https://github.com/ipfs/kubo/pull/10084 and looks green :+1:

**Review by hacdias — APPROVED** [review]

File: .github/workflows/gateway-conformance.yml (hunk 2 above)
Comment (hacdias) [inline .github/workflows/gateway-conformance.yml]:
```suggestion
        uses: ipfs/gateway-conformance/.github/actions/test@v0.3
```

**Pull request merged**